mllib/src/main/scala/org/apache/spark/ml/tree/Split.scala

```diff
@@ -189,3 +189,76 @@ class ContinuousSplit private[ml] (override val featureIndex: Int, val threshold
     OldSplit(featureIndex, threshold, OldFeatureType.Continuous, List.empty[Double])
   }
 }
+
+/**
+ * Split with pointers to the left and right child nodes in decision tree.
+ * Used to update the node index cache.
+ * @param split Input split
+ */
+class SplitWithChildNodeInfo private[ml] (split: Split) extends Split {
+  private var _leftChildIndex : Option[Int] = None
+  private var _rightChildIndex : Option[Int] = None
```

Comment:
I was following the convention that they used in some of the other files. I can change it.